v2/args/args.go

```diff
@@ -137,7 +137,7 @@ func (g *GeneratorArgs) NewBuilder() (*parser.Builder, error) {
 // Execute implements main().
 // If you don't need any non-default behavior, use as:
 // args.Default().Execute(...)
-func (g *GeneratorArgs) Execute(nameSystems namer.NameSystems, defaultSystem string, pkgs func(*generator.Context, *GeneratorArgs) []generator.Package) error {
+func (g *GeneratorArgs) Execute(nameSystems namer.NameSystems, defaultSystem string, getTargets func(*generator.Context, *GeneratorArgs) []generator.Target) error {
```

Comment:
If the goal is to distinguish input and output packages, how about making that explicit in the type names? `InputPackage` / `OutputPackage`, or `InPackage` / `OutPackage`, or `InputPkg` etc.

Reply:
I was reviewing this in the context of https://github.com/kubernetes/gengo/pull/259/commits/da029b0c88feccc77a402942cda0e7782a97dbdd — the commit message says

> Having two types names "Package" was super confusing to read.  I don't love "Target" but at least it implies "this is the output" rather than the input.

Since the use of `Package` is ambiguous, it seems to me that a possible solution would be to disambiguate both uses cases instead of using a not-great replacement for the output scenario; `InputPackage` / `OutputPackage` for example.

Reply:
Oh, I see.  I was shooting for something shortish rather than renaming `types.Package` at this time.  In truth, I rather hate the way gengo has its own `Package` type with (IMO) dangerous lazy-load semantics and how it interposes its own `Type` and `Kind` types with different semantics from Go.  I'd like to see use eliminate all of that, but it wasn't necessary to this PR and I *really* want to get workspaces support into k/k (which is blocked on this :)

So - we could totally do a followup to rename things more :)

Reply:
That’s fine by me!